assert/_diff.ts

```diff
@@ -201,7 +203,15 @@ export function diff<T>(A: T[], B: T[]): Array<DiffResult<T>> {
     return fp;
   }
 
-  while (fp[delta + offset].y < N) {
+  function ensureDefined<T>(item: T | undefined): T {
+    if (!item) {
+      throw Error("Unexpected missing FarthestPoint");
+    }
+    return item;
+  }
```

Comment:
```suggestion
  function ensureDefined<T>(item?: T): T {
    if (item === undefined) {
      throw Error("Unexpected missing FarthestPoint");
    }
    return item;
  }
```
For strictness. Also, I don't see a reason for this to be a nested function. Can we please make it top-level?

Reply:
I did this following convention in this file, the `diff` function has 3 nested functions. Happy to move it though.